pkg/plugins/manager/ownership_unix.go

```diff
@@ -0,0 +1,18 @@
+//go:build !windows
+
+package manager
+
+import (
+    "os"
+    "syscall"
+)
+
+// matchOwnershipToParent ensures child inherits ownership from parent dir
+func matchOwnershipToParent(child, parent string) error {
+    parentInfo, err := os.Stat(parent)
+    if err != nil {
+        return err
+    }
+    stat := parentInfo.Sys().(*syscall.Stat_t)
```

Comment:
i haven't dug too deeply into the go source to know what other build targets would match `!windows`, but it seems like it's probably worth moving from a type assertion to testing for the expected type here to avoid panics:
```suggestion
    stat, ok := parentInfo.Sys().(*syscall.Stat_t)
    if !ok {
      // return error here
    }
```

@wbrowne does this approach seem valid, or would it be better to handle this during [extraction](https://github.com/grafana/grafana/blob/2a5ba2e74a8954a82393c593f2987fac4d40d5f0/pkg/plugins/storage/fs.go#L68) handling file system operations?